Marlin/src/module/endstops.cpp

```diff
@@ -554,20 +554,16 @@ void Endstops::update() {
       endstop_poll_count = 7;
       old_live_state = live_state;
     }
-    else if (endstop_poll_count && !--endstop_poll_count)
-      validated_live_state = live_state;
-
-  #else
+    else if (endstop_poll_count && --endstop_poll_count) return;
```

Comment:
@ejtagle — The only other big change here. If the state hasn't been validated then don't even continue on with the rest of the method.

Reply:
This specific change is right. If state was not validated... No, i am not sure. The logic is sound, but Marlin counts on endstop_triggered() being call at all times ... Otherwise blocks will not be cancelled if forcing an endstop...